alf/algorithms/generator.py

```diff
@@ -119,47 +119,105 @@ def predict_step(self, inputs, state=None, requires_jac_diag=False):
 @alf.configurable
 class InverseMVPAlgorithm(Algorithm):
     r"""InverseMVP network Algorithm
-        
-    Simple MLP network used with the functional gradient par_vi methods
-    It is used to predict :math:`x=J^{-1}*eps` given eps for the purpose of 
-    optimizing a downstream objective Jx - eps = 0. 
-    
-    If using ``svgd3``, then the eps quantity represents the kernel grad
-        :math:`\nabla_{z'}k(z', z)`
+
+    Maintain an encoding network that takes (z, vec) as input and predicts a 
+    matrix-vector product (mvp) of the form :math:`y=J^{-1}(z)*vec`, where 
+    :math:`J^{-1}(z)` is the inverse of the Jacobian matrix of some function
+    :math:`f(z)`, and ``vec`` is a vector. This network is used in GPVI in 
+    computing the ``functional_gradient`` of the generator, where :math:`J^{-1}`
+    is the inverse of the Jacobian of the generator function w.r.t. input noise
+    :math:`z'`, and ``vec`` is the gradient of the kernel 
+    :math:`\nabla_{z'}k(z', z)`.
+
+    Training of this network is done outside of the algorithm, where the network is 
+    trained to predict :math:`y` that minimize the  objective :math:`||Jy - vec||^2.
     """
 
     def __init__(self,
+                 input_dim,
                  output_dim,
-                 net: Network,
+                 hidden_size=100,
+                 num_hidden_layers=1,
+                 activation=torch.relu_,
                  optimizer=None,
                  name="InverseMVPAlgorithm"):
         r"""Create a InverseMVPAlgorithm.
         Args:
-            output_dim (int): total output dimension of the inverse_mvp net, will differ
-                between ``svgd3`` and ``minmax`` methods
-            net (Network): network for predicting outputs from inputs.
+            input_dim (int): dimension of input z
+            output_dim (int): output dimension, i.e., dimension of the mvp 
+            hidden_size (int): width of hidden layers
+            num_hidden_layers (int): number of hidden layers after 
+            activation (nn.functional): activation used for all hidden layers.
```

Comment:
nn.functional => `Callable`. nn.functional is not a type.

Reply:
Updated.